### PR TITLE
[Merged by Bors] - chore: pi_bounds cleanup, add mathematica script

### PR DESCRIPTION
**Pull request by digama0**
* Rewrites the `pi_(lower|upper)_bound` tactics to allow some more general input
* Clean up outdated porting notes
* Add a mathematica script so people can go further if they want
* Rename the pi bound theorems to have names like `pi_gt_d6` since the existing names do not scale well
  * `pi_gt_314` -> `pi_gt_d2`
  * `pi_lt_315` -> `pi_lt_d2`
  * `pi_gt_31415` -> `pi_gt_d4`
  * `pi_lt_31416` -> `pi_lt_d4`
  * `pi_gt_3141592` -> `pi_gt_d6`
  * `pi_lt_3141593` -> `pi_lt_d6`
* Add a 20-digit calculation of pi [on request](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Code.20to.20generate.20the.20pi.20bound.20witnesses/near/470184223) since this file is not slow anymore

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Command-Master** [comment]
Could you add the renames to the commit message?

File: Mathlib/Data/Real/Pi/Bounds.lean [hunk 1]
```diff
@@ -17,28 +17,23 @@ See also `Mathlib/Data/Real/Pi/Leibniz.lean` and `Mathlib/Data/Real/Pi/Wallis.le
 formulas for `π`.
 -/
```
Comment (eric-wieser) [inline Mathlib/Data/Real/Pi/Bounds.lean:18]: Can you fix the theorem names in the docstring?

File: Mathlib/Data/Real/Pi/Bounds.lean [hunk 2]
```diff
@@ -91,28 +84,19 @@ section Tactic
...
-  let n := ← (toExpr ratStx.size).toSyntax
-  let els := (ratStx.map numDen).reduceOption
+  have els := l.getElems
+  let n := quote els.size
   evalTactic (← `(tactic| apply pi_lower_bound_start $n))
-  let _ := ← els.mapM fun (x, y) => do
-      evalTactic (← `(tactic| apply sqrtTwoAddSeries_step_up $x $y))
+  for l in els do
+    let Q := .const ``_root_.Rat []
+    let {num, den, ..} ← unsafe Meta.evalExpr ℚ Q (← Term.elabTermAndSynthesize l (some Q))
```
Comment (eric-wieser) [inline Mathlib/Data/Real/Pi/Bounds.lean:96]: nit: could write
```suggestion
    let {num, den, ..} ← unsafe Meta.evalExpr ℚ q(ℚ) (← Term.elabTermAndSynthesize l (some q(ℚ)))
```

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/Data/Real/Pi/Bounds.lean [hunk 3]
```diff
@@ -141,52 +105,133 @@ theorem sqrtTwoAddSeries_step_down (a b : ℕ) {c d n : ℕ} {z : ℝ}
...
+@[deprecated (since := "2024-09-19")] alias pi_gt_3141592 := pi_gt_d6
+
+theorem pi_lt_d6 : π < 3.141593 := by
+  -- bound[3141593*^-6, Iters -> 11, Rounding -> .5, Precision -> 17]
+  pi_upper_bound [
+    35839/25342, 49143/26596, 145729/74292, 294095/147759, 2-137/56868, 2-471/781921, 2-153/1015961,
+    2-157/4170049, 2-28/2974805, 2-9/3824747, 2-7/11899211]
+@[deprecated (since := "2024-09-19")] alias pi_lt_3141593 := pi_lt_d6
+
+theorem pi_gt_d20 : 3.14159265358979323846 < π  := by
```
Comment (vihdzp) [inline Mathlib/Data/Real/Pi/Bounds.lean:205]:
```suggestion
theorem pi_gt_d20 : 3.14159265358979323846 < π := by
```

**Pull request closed without merging**